### ISSUE TITLE
Use a published Antora sitemap.xml for autocomplete/validation of links

### ISSUE THREAD

**Issue by ahus1** [asciidoctor/asciidoctor-intellij-plugin#454]
Given a base URL, any xref with a version could link to the site's currently rendered page. A minimal antora-playbook.yml could be placed in the current project containing that piece of information. Usually there would be a separate publishing project, but I found it good for local testing to have minimal playbook around for the current for local testing.

I found that Antora already creates a sitemap. Auto-complete could work based on the sitemap (see [Antora's sitemap for the antora component](https://docs.antora.org/sitemap-antora.xml) as an example); the logic would be [baseurl]/sitemap-[component].xml. Each URL in the sitemap could be parsed as [baseurl]/[component]/[module]/[page]. Different versions/configurations might either create a path or a file ending with .html; see Camel for a different example (https://camel.apache.org/sitemap-manual.xml).

It could also try to look up the sitemaps from a local output dir of Antora. 

IMHO this should be sufficient for a first prototype. Once Antora provides more metadata, this can then be enhanced.

This approach would solve issue when linking to other versions and components not part of the user's current Git repo.

_Originally posted by @ahus1 in https://github.com/asciidoctor/asciidoctor-intellij-plugin/issues/377#issuecomment-596784901_

**Comment by mojavelinux** [comment]
> Once Antora provides more metadata, this can then be enhanced.

Indeed, Antora is planning to provide a supplemental manifest file that exports a lot more information about the published site and the files it contains than what the sitemap.xml allows. Something similar to https://developer.chrome.com/extensions/manifest, but likely tailored for Antora.

**Comment by ahus1** [comment]
This now probably superseded by #1250 as the Antora site manifest includes also the titles of the pages. Please comment if  someone see this differently.
